### PR TITLE
Refactor `ObjectClient.get_object` to use an `GetObjectParams` parameter

### PR DESCRIPTION
**Pull request by muddyfish**
## Description of change

Refactor `ObjectClient.get_object` to use an `&GetObjectParams` parameter.

Migrates the two existing parameters, `range` and `if_match` to `GetObjectParams` and changes all call sites.

Relevant issues: N/A

## Does this change impact existing behavior?

No

## Does this change need a changelog entry in any of the crates?

Yes. Breaking change in mountpoint-s3-client. 

---

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and I agree to the terms of the [Developer Certificate of Origin (DCO)](https://developercertificate.org/).

### CODE REVIEW

File: mountpoint-s3-client/src/object_client.rs [hunk 1]
```diff
@@ -78,8 +78,7 @@ pub trait ObjectClient {
         &self,
         bucket: &str,
         key: &str,
-        range: Option<Range<u64>>,
-        if_match: Option<ETag>,
+        params: &GetObjectParams,
```
Comment (vladem) [inline mountpoint-s3-client/src/object_client.rs:81]: a breaking change, a Changelog entry is required

File: mountpoint-s3-client/src/failure_client.rs [hunk 2]
```diff
@@ -123,17 +123,16 @@ where
...
-        if_match: Option<ETag>,
+        params: &GetObjectParams,
     ) -> ObjectClientResult<Self::GetObjectRequest, GetObjectError, Self::ClientError> {
         let wrapper = (self.get_object_cb)(
             &mut *self.state.lock().unwrap(),
             bucket,
             key,
-            range.clone(),
-            if_match.clone(),
+            params.range.clone(),
```
Comment (vladem) [inline mountpoint-s3-client/src/failure_client.rs:132]: may `get_object_cb` accept `&GetObjectParams` as-is?

File: mountpoint-s3-client/src/s3_crt_client.rs [hunk 3]
```diff
@@ -1281,12 +1281,9 @@ impl ObjectClient for S3CrtClient {
         &self,
         bucket: &str,
         key: &str,
-        range: Option<Range<u64>>,
-        if_match: Option<ETag>,
-        // TODO: If more arguments are added to get object, make a request struct having those arguments
```
Comment (vladem) [inline mountpoint-s3-client/src/s3_crt_client.rs:1286]: nice!

File: mountpoint-s3-client/src/s3_crt_client/get_object.rs [hunk 4]
```diff
@@ -35,10 +36,9 @@ impl S3CrtClient {
         &self,
         bucket: &str,
         key: &str,
-        range: Option<Range<u64>>,
-        if_match: Option<ETag>,
+        params: &GetObjectParams,
     ) -> Result<S3GetObjectRequest, ObjectClientError<GetObjectError, S3RequestError>> {
-        let span = request_span!(self.inner, "get_object", bucket, key, ?range, ?if_match);
+        let span = request_span!(self.inner, "get_object", bucket, key, ?params.range, ?params.if_match);
```
Comment (vladem) [inline mountpoint-s3-client/src/s3_crt_client/get_object.rs:41]:
```suggestion
        let span = request_span!(self.inner, "get_object", bucket, key, range=?params.range, if_match=?params.if_match);
```

**Review by dannycjones — APPROVED** [review]
LGTM!

**Pull request merged**